src/redux/stores/balances/balancesApi.ts

```diff
@@ -91,3 +103,64 @@ export const getTransactionsLocalStorageKey: (
   walletAddress: string,
   chainId: number
 ) => string = (walletAddress, chainId) => `airswap-marketplace/transactions/${walletAddress}/${chainId}`;
+
+export const fetchTokenIds = createAsyncThunk<number[], fetchTokenIdsParams>(
+  'balances/fetchTokenIds',
+  async ({
+    provider,
+    walletAddress,
+    collectionTokenAddress,
+  }) => {
+    // 0x780e9d63 is the interface ID for erc721 enumerable
+    const ERC721Enumerable = '0x780e9d63';
+
+    const contract = new ethers.Contract(collectionTokenAddress, ERC165_ABI, provider);
+
+    const isERC721Enumerable = await contract.supportsInterface(ERC721Enumerable);
+    if (isERC721Enumerable) {
+      const collectionContract = new ethers.Contract(collectionTokenAddress, ERC721Enumerable_ABI, provider);
```

Comment:
```suggestion
      const collectionContract = new ethers.Contract(collectionToken.address, erc721EnumerableContract.abi, provider);
```

Same for other tokens.